### PR TITLE
Some miscellaneous separated/stable type facts

### PR DESCRIPTION
**Pull request by dolio**
Added separatedness of `Bool`

Added some convenient functions for deriving `Stable` and `Separated` for e.g. functions.

These were some general facts I found useful while working on some other stuff, sort of similar to various h-level combinators.

### CODE REVIEW

**Review by MatthiasHu — CHANGES_REQUESTED** [review]
Nice contribution!
I have some remarks, I hope they are not too annoying. :-)

File: Cubical/Relation/Nullary/Properties.agda [hunk 1]
```diff
@@ -48,6 +49,17 @@ Stable¬ ¬¬¬a a = ¬¬¬a ¬¬a
   where
   ¬¬a = λ ¬a → ¬a a
 
+Stable→ : Stable B → Stable (A → B)
+Stable→ Bs e x = Bs λ k → e λ f → k (f x)
+
+StableΠ : (∀ x → Stable (P x)) -> Stable (∀ x → P x)
+StableΠ Ps e x = Ps x λ k → e λ f → k (f x)
+
+Stable× : Stable A -> Stable B -> Stable (Σ A (λ _ → B))
```
[the remark this answers is not in the document]
Comment (dolio) [reply Cubical/Relation/Nullary/Properties.agda]: I see. I didn't realize the `.Base` module exported `_×_`.

File: Cubical/Relation/Nullary/Properties.agda [hunk 2]
```diff
@@ -48,6 +49,17 @@ Stable¬ ¬¬¬a a = ¬¬¬a ¬¬a
   where
   ¬¬a = λ ¬a → ¬a a
 
+Stable→ : Stable B → Stable (A → B)
+Stable→ Bs e x = Bs λ k → e λ f → k (f x)
```
[the remark this answers is not in the document]
Comment (dolio) [reply Cubical/Relation/Nullary/Properties.agda]: I think I just proved the `→` versions first, before I was sure that the `Π` version would work.

**Review by MatthiasHu — APPROVED** [review]

**Comment by MatthiasHu** [comment]
I will merge this when the CI is done.

**Pull request merged**